runtime/wasm/src/host.rs

```diff
@@ -106,15 +113,25 @@ impl RuntimeHost {
             let module = WasmiModule::new(&logger, wasmi_config);
 
             // Process one event at a time, blocking the thread when waiting for
-            // the next event.
+            // the next event. Also check for a cancelation signal.
             Self::subscribe_to_events(&logger, data_source, module, ethereum_adapter)
+                .select(
```

Comment:
Yes, our first use (I believe) of `select`. Select is beautiful.

Reply:
There's another use in this file, when we select among all ethereum event subscriptions. Beautiful it is.